backtester/config/config.go

```diff
@@ -112,73 +140,177 @@ func (c *Config) PrintSetting() {
 	log.Info(log.BackTester, "-------------------------------------------------------------\n\n")
 }
 
-// Validate ensures no one sets bad config values on purpose
-func (m *MinMax) Validate() {
-	if m.MaximumSize < 0 {
-		m.MaximumSize *= -1
-		log.Warnf(log.BackTester, "invalid maximum size set to %f", m.MaximumSize)
+// Validate checks all config settings
+func (c *Config) Validate() error {
+	err := c.validateDate()
+	if err != nil {
+		return err
+	}
+	err = c.validateStrategySettings()
+	if err != nil {
+		return err
+	}
+	err = c.validateCurrencySettings()
+	if err != nil {
+		return err
+	}
+	return c.validateMinMaxes()
+}
+
+// validate ensures no one sets bad config values on purpose
+func (m *MinMax) validate() error {
+	if m.MaximumSize.IsNegative() {
+		return fmt.Errorf("invalid maximum size %w", errSizeLessThanZero)
 	}
-	if m.MinimumSize < 0 {
-		m.MinimumSize *= -1
-		log.Warnf(log.BackTester, "invalid minimum size set to %f", m.MinimumSize)
+	if m.MinimumSize.IsNegative() {
+		return fmt.Errorf("invalid minimum size %w", errSizeLessThanZero)
 	}
-	if m.MaximumSize <= m.MinimumSize && m.MinimumSize != 0 && m.MaximumSize != 0 {
-		m.MaximumSize = m.MinimumSize + 1
-		log.Warnf(log.BackTester, "invalid maximum size set to %f", m.MaximumSize)
+	if m.MaximumTotal.IsNegative() {
+		return fmt.Errorf("invalid maximum total set to %w", errSizeLessThanZero)
 	}
-	if m.MaximumTotal < 0 {
-		m.MaximumTotal *= -1
-		log.Warnf(log.BackTester, "invalid maximum total set to %f", m.MaximumTotal)
+	if m.MaximumSize.LessThanOrEqual(m.MinimumSize) && !m.MinimumSize.IsZero() && !m.MaximumSize.IsZero() {
```

Comment:
After investigating this + you latest review, I'm now thinking that this would need to become its own separate feature, if requested. Order size limits are done with fees in mind to prevent overspend. So if you have a max order amount set to 0.4 for example, the backtester will size the order so that you spend 0.4 worth of the currency inclusive of fees. Having a max and min of the same value, along with all the sizing and resizing that occurs results in a number that is less than the minimum sized amount. 

The portfolio manager will say, you have 0.4 BTC worth of USDT to spend on the order and fees, then the exchange manager will resize the order amount based on exchange limits, candle size, portfolio spending limits and slippage rates which will put the amount ordered under the amount available to it which causes no orders to be placed when there is zero wiggle room allowed.

This is a protection measure for say, if the available funds is now at 0.009, then don't bother with orders, rather than a feature to allow for you to buy only 0.4 BTC per dataEvent.

This sounds like a feature flag to set limits on quantity only which is new functionality and I find this feature to be outside the scope of this PR. Though I'd be happy to hear feedback on what I've just outlined and whether you have an alternative solution. For now, I'll be reverting this `LessThanOrEqual` change

Reply:
After looking at the current logic I agree with that assessment and should not be included in this PR. Thanks for the detailed explanation.